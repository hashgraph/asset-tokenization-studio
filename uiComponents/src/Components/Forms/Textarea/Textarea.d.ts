import type { SystemStyleObject as ChakraSystemStyleConfig, ComponentWithAs } from "@chakra-ui/system";
import { type TextareaProps as ChakraTextareaProps } from "@chakra-ui/textarea";
import type { BaseMultiStyleConfiguration, PartsStyleInterpolation } from "@Theme/Components/BaseMultiStyleConfiguration";
export declare const textareaPartsList: Array<"labelContainer" | "label" | "container" | "length">;
type Parts = typeof textareaPartsList;
export type TextareaConfigProps = {
    variant: TextareaProps["variant"];
    isInvalid?: boolean;
    isSuccess?: boolean;
    isDisabled?: boolean;
    hasLabel?: boolean;
};
export interface TextareaThemeConfiguration extends BaseMultiStyleConfiguration<Parts> {
    baseStyle: (({ variant, isInvalid, isDisabled, }: TextareaConfigProps) => Partial<Record<Parts[number], ChakraSystemStyleConfig>>) | PartsStyleInterpolation<Parts>;
}
export interface TextareaProps extends ChakraTextareaProps {
    isSuccess?: boolean;
    label?: string;
    showRequired?: boolean;
}
export declare const Textarea: ComponentWithAs<"textarea", TextareaProps>;
export {};
