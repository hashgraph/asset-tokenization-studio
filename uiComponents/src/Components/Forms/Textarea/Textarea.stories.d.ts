import React from "react";
declare const meta: import("@storybook/types").ComponentAnnotations<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.TextareaHTMLAttributes<HTMLTextAreaElement>, HTMLTextAreaElement>, any, import("./Textarea").TextareaProps, import("@chakra-ui/system/dist/system.types").As>>;
export default meta;
export declare const Default: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.TextareaHTMLAttributes<HTMLTextAreaElement>, HTMLTextAreaElement>, any, import("./Textarea").TextareaProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const Disabled: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.TextareaHTMLAttributes<HTMLTextAreaElement>, HTMLTextAreaElement>, any, import("./Textarea").TextareaProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const Invalid: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.TextareaHTMLAttributes<HTMLTextAreaElement>, HTMLTextAreaElement>, any, import("./Textarea").TextareaProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const Valid: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.TextareaHTMLAttributes<HTMLTextAreaElement>, HTMLTextAreaElement>, any, import("./Textarea").TextareaProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const WithMaxLength: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.TextareaHTMLAttributes<HTMLTextAreaElement>, HTMLTextAreaElement>, any, import("./Textarea").TextareaProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const ControlledTextarea: () => JSX.Element;
