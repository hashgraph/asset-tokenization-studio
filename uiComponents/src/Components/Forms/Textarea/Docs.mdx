import {
  Story,
  Meta,
  Title,
  ArgTypes,
  Primary,
  Stories,
  Source,
} from "@storybook/addon-docs";
import { Grid, GridItem, Text, Link, Center, Flex } from "@chakra-ui/react";
import * as TextareaStories from "./Textarea.stories";

<Meta of={TextareaStories} />

<Title />

<a
  href="https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/NEW---ioBricks-Library?node-id=1503%3A22829&t=CYTpHR0mFDJMv5GO-4"
  target="\_blank"
>
  View figma design
</a>

<Primary />

## How to set up

### Config theme

Add the following configurations to the `theme.tsx` and `Textarea.ts` files, respectively.

```tsx dark
const theme = {
  components: {
    Textarea: ConfigTextarea,
  },
};
```

```tsx dark
const ConfigTextarea: TextareaThemeConfiguration = {
  parts: textareaPartsList,
  baseStyle: {},
};
```

### Component parts

You have these parts available: `"labelContainer" | "label" | "container" | "length"` (`textareaPartsList`)

## Props

<ArgTypes of={TextareaStories} />

## Stories

<Stories title="" />
