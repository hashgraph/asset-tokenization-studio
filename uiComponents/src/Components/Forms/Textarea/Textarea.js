var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import { forwardRef, useMultiStyleConfig as useChakraMultiStyleConfig, } from "@chakra-ui/system";
import { FormLabel as ChakraFormLabel } from "@chakra-ui/form-control";
import { Textarea as ChakraTextarea, } from "@chakra-ui/textarea";
import React from "react";
import { Text } from "@/Components/Foundations/Text";
import { Box } from "@chakra-ui/react";
export var textareaPartsList = ["container", "label", "labelContainer", "length"];
export var Textarea = forwardRef(function (_a, ref) {
    var size = _a.size, variant = _a.variant, isInvalid = _a.isInvalid, isSuccess = _a.isSuccess, isDisabled = _a.isDisabled, isRequired = _a.isRequired, label = _a.label, _b = _a.showRequired, showRequired = _b === void 0 ? true : _b, id = _a.id, value = _a.value, maxLength = _a.maxLength, props = __rest(_a, ["size", "variant", "isInvalid", "isSuccess", "isDisabled", "isRequired", "label", "showRequired", "id", "value", "maxLength"]);
    var styles = useChakraMultiStyleConfig("Textarea", {
        size: size,
        variant: variant,
        isInvalid: isInvalid,
        isSuccess: isSuccess,
        isDisabled: isDisabled,
        hasLabel: Boolean(label)
    });
    return (React.createElement(ChakraFormLabel, { sx: styles.labelContainer, htmlFor: id, flex: 1, position: "relative", ref: ref },
        label && (React.createElement(Text, { as: "span", id: "label", display: "flex", sx: styles.label },
            label,
            showRequired && isRequired && "*")),
        React.createElement(Box, { pos: "relative" },
            React.createElement(ChakraTextarea, __assign({ variant: variant, size: size, ref: ref, id: id, isDisabled: isDisabled, isInvalid: isInvalid, value: value, maxLength: maxLength, sx: styles.container }, props)),
            maxLength && (React.createElement(Text, { sx: styles.length },
                (value === null || value === void 0 ? void 0 : value.toString().length) || 0,
                "/",
                maxLength)))));
});
