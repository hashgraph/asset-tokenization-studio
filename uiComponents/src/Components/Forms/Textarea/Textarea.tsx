import type {
  SystemStyleObject as ChakraSystemStyleConfig,
  ComponentWithAs,
} from "@chakra-ui/system";
import {
  forwardRef,
  useMultiStyleConfig as useChakraMultiStyleConfig,
} from "@chakra-ui/system";
import { FormLabel as ChakraFormLabel } from "@chakra-ui/form-control";
import {
  Textarea as ChakraTextarea,
  type TextareaProps as ChakraTextareaProps,
} from "@chakra-ui/textarea";
import type {
  BaseMultiStyleConfiguration,
  PartsStyleInterpolation,
} from "@Theme/Components/BaseMultiStyleConfiguration";
import React from "react";
import { Text } from "@/Components/Foundations/Text";
import { Box } from "@chakra-ui/react";

export const textareaPartsList: Array<
  "labelContainer" | "label" | "container" | "length"
> = ["container", "label", "labelContainer", "length"];

type Parts = typeof textareaPartsList;

export type TextareaConfigProps = {
  variant: TextareaProps["variant"];
  isInvalid?: boolean;
  isSuccess?: boolean;
  isDisabled?: boolean;
  hasLabel?: boolean;
};
export interface TextareaThemeConfiguration
  extends BaseMultiStyleConfiguration<Parts> {
  baseStyle:
    | (({
        variant,
        isInvalid,
        isDisabled,
      }: TextareaConfigProps) => Partial<
        Record<Parts[number], ChakraSystemStyleConfig>
      >)
    | PartsStyleInterpolation<Parts>;
}

export interface TextareaProps extends ChakraTextareaProps {
  isSuccess?: boolean;
  label?: string;
  showRequired?: boolean;
}

export const Textarea: ComponentWithAs<"textarea", TextareaProps> = forwardRef<
  TextareaProps,
  "textarea"
>(
  (
    {
      size,
      variant,
      isInvalid,
      isSuccess,
      isDisabled,
      isRequired,
      label,
      showRequired = true,
      id,
      value,
      maxLength,
      ...props
    }: TextareaProps,
    ref
  ) => {
    const styles = useChakraMultiStyleConfig("Textarea", {
      size,
      variant,
      isInvalid,
      isSuccess,
      isDisabled,
      hasLabel: Boolean(label),
    });

    return (
      <ChakraFormLabel
        sx={styles.labelContainer}
        htmlFor={id}
        flex={1}
        position="relative"
        ref={ref}
      >
        {label && (
          <Text as="span" id="label" display="flex" sx={styles.label}>
            {label}
            {showRequired && isRequired && "*"}
          </Text>
        )}
        <Box pos="relative">
          <ChakraTextarea
            variant={variant}
            size={size}
            ref={ref}
            id={id}
            isDisabled={isDisabled}
            isInvalid={isInvalid}
            value={value}
            maxLength={maxLength}
            sx={styles.container}
            {...props}
          />
          {maxLength && (
            <Text sx={styles.length}>
              {value?.toString().length || 0}/{maxLength}
            </Text>
          )}
        </Box>
      </ChakraFormLabel>
    );
  }
);
