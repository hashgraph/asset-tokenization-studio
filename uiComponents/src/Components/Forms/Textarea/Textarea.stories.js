var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import { Button } from "@Components/Interaction/Button";
import { linkTo } from "@storybook/addon-links";
import React from "react";
import { Textarea } from "./Textarea";
var meta = {
    title: "Design System/Forms/Textarea",
    component: Textarea,
    args: {
        placeholder: "Placeholder",
        label: "Label"
    },
    argTypes: {
        showRequired: {
            control: { type: "boolean" },
            description: "Boolean that toggles whether to show the * if it is required."
        },
        isRequired: {
            control: { type: "boolean" },
            description: "Boolean to specify that the input is required."
        },
        isDisabled: {
            control: { type: "boolean" },
            description: "Boolean to specify if the input is disabled."
        },
        isSuccess: {
            control: { type: "boolean" },
            description: "Boolean to specify that the input is valid and success."
        },
        placeholder: {
            description: "Placeholder of the input."
        },
        label: {
            description: "Label of the input."
        },
        variant: {
            control: false,
            description: "Variant of the input. Must be defined in the theme."
        }
    },
    parameters: {
        design: {
            type: "figma",
            url: "https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/NEW---ioBricks-Library?node-id=1503%3A22829&t=CYTpHR0mFDJMv5GO-4"
        },
        docs: {}
    }
};
export default meta;
var Template = function (args) { return React.createElement(Textarea, __assign({}, args)); };
export var Default = Template.bind({});
export var Disabled = Template.bind({});
Disabled.args = {
    isDisabled: true
};
export var Invalid = Template.bind({});
Invalid.args = {
    isInvalid: true
};
export var Valid = Template.bind({});
Valid.args = {
    isSuccess: true
};
export var WithMaxLength = Template.bind({});
WithMaxLength.args = {
    maxLength: 100
};
export var ControlledTextarea = function () { return (React.createElement(Button, { onClick: linkTo("Design System/Forms/Controllers/TextareaController") }, "Check out the TextareaController component stories")); };
