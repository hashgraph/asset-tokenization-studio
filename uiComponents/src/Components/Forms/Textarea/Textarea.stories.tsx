import { Button } from "@Components/Interaction/Button";

import { linkTo } from "@storybook/addon-links";
import type { Meta, StoryFn } from "@storybook/react";
import React from "react";
import { Textarea } from "./Textarea";

const meta = {
  title: "Design System/Forms/Textarea",
  component: Textarea,
  args: {
    placeholder: "Placeholder",
    label: "Label",
  },
  argTypes: {
    showRequired: {
      control: { type: "boolean" },
      description:
        "Boolean that toggles whether to show the * if it is required.",
    },
    isRequired: {
      control: { type: "boolean" },
      description: "Boolean to specify that the input is required.",
    },
    isDisabled: {
      control: { type: "boolean" },
      description: "Boolean to specify if the input is disabled.",
    },
    isSuccess: {
      control: { type: "boolean" },
      description: "Boolean to specify that the input is valid and success.",
    },
    placeholder: {
      description: "Placeholder of the input.",
    },
    label: {
      description: "Label of the input.",
    },
    variant: {
      control: false,
      description: "Variant of the input. Must be defined in the theme.",
    },
  },
  parameters: {
    design: {
      type: "figma",
      url: "https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/NEW---ioBricks-Library?node-id=1503%3A22829&t=CYTpHR0mFDJMv5GO-4",
    },
    docs: {},
  },
} as Meta<typeof Textarea>;
export default meta;

const Template: StoryFn<typeof Textarea> = (args) => <Textarea {...args} />;

export const Default = Template.bind({});

export const Disabled = Template.bind({});

Disabled.args = {
  isDisabled: true,
};

export const Invalid = Template.bind({});

Invalid.args = {
  isInvalid: true,
};

export const Valid = Template.bind({});

Valid.args = {
  isSuccess: true,
};

export const WithMaxLength = Template.bind({});

WithMaxLength.args = {
  maxLength: 100,
};

export const ControlledTextarea = () => (
  <Button
    onClick={linkTo("Design System/Forms/Controllers/TextareaController")}
  >
    Check out the TextareaController component stories
  </Button>
);
