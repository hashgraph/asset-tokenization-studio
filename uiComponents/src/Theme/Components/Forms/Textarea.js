import { textareaPartsList, } from "@Components/Forms/Textarea";
export var ConfigTextarea = {
    parts: textareaPartsList,
    baseStyle: function () { return ({
        labelContainer: {
            width: "100%",
            position: "relative"
        },
        label: {
            apply: "textStyles.ElementsRegularSM",
            color: "neutral.700",
            zIndex: 1,
            mb: 2
        },
        container: {
            fontWeight: "normal",
            apply: "textStyles.ElementsRegularSM",
            color: "neutral.800",
            _placeholder: {
                apply: "textStyles.ElementsRegularSM",
                color: "neutral.400"
            }
        },
        length: {
            position: "absolute",
            right: 3,
            apply: "textStyles.ElementsRegularXS",
            bottom: 2,
            color: "neutral.600"
        }
    }); },
    variants: {
        outline: function (_a) {
            var isSuccess = _a.isSuccess;
            return ({
                container: {
                    paddingX: 3,
                    paddingY: 2,
                    borderRadius: "simple",
                    border: "1px",
                    borderColor: isSuccess ? "success" : "neutral.300",
                    _hover: {
                        borderColor: "neutral.500"
                    },
                    _focus: {
                        borderColor: isSuccess ? "success" : "primary",
                        _hover: { borderColor: isSuccess ? "success" : "primary" }
                    },
                    _focusVisible: { boxShadow: "none" },
                    _invalid: {
                        borderColor: "error",
                        boxShadow: "none",
                        _focusVisible: {
                            borderColor: "error"
                        }
                    },
                    _disabled: {
                        borderColor: "neutral.100",
                        color: "neutral.300",
                        bg: "neutral.50"
                    }
                }
            });
        }
    },
    defaultProps: {
        variant: "outline"
    }
};
