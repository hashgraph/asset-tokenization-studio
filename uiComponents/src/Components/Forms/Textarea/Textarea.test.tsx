import { render } from "@testing-library/react";
import React from "react";
import { Textarea } from "./Textarea";

describe(`<Textarea />`, () => {
  const testId = "textareaTest";

  const defaultProps = {};

  const factoryComponent = (props = defaultProps) =>
    render(<Textarea data-testid={testId} {...props} />);

  test("renders correctly", () => {
    const component = factoryComponent();
    expect(component.asFragment()).toMatchSnapshot();
  });

  test("shows correctly when is disabled", () => {
    const component = factoryComponent({ isDisabled: true });
    expect(component.asFragment()).toMatchSnapshot("Using disabled");
  });

  test("shows correctly when is invalid", () => {
    const component = factoryComponent({ isInvalid: true });
    expect(component.asFragment()).toMatchSnapshot("Using invalid");
  });

  test("shows correctly when is success", () => {
    const component = factoryComponent({ isSuccess: true });
    expect(component.asFragment()).toMatchSnapshot("Using success");
  });
});
