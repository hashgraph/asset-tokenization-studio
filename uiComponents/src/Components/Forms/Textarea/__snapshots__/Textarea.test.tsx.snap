// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Textarea /> renders correctly 1`] = `
<DocumentFragment>
  <label
    class="chakra-form__label css-tigm22"
  >
    <div
      class="css-79elbk"
    >
      <textarea
        class="chakra-textarea css-0"
        data-testid="textareaTest"
      />
    </div>
  </label>
</DocumentFragment>
`;

exports[`<Textarea /> shows correctly when is disabled: Using disabled 1`] = `
<DocumentFragment>
  <label
    class="chakra-form__label css-tigm22"
  >
    <div
      class="css-79elbk"
    >
      <textarea
        class="chakra-textarea css-0"
        data-testid="textareaTest"
        disabled=""
      />
    </div>
  </label>
</DocumentFragment>
`;

exports[`<Textarea /> shows correctly when is invalid: Using invalid 1`] = `
<DocumentFragment>
  <label
    class="chakra-form__label css-tigm22"
  >
    <div
      class="css-79elbk"
    >
      <textarea
        aria-invalid="true"
        class="chakra-textarea css-0"
        data-testid="textareaTest"
      />
    </div>
  </label>
</DocumentFragment>
`;

exports[`<Textarea /> shows correctly when is success: Using success 1`] = `
<DocumentFragment>
  <label
    class="chakra-form__label css-tigm22"
  >
    <div
      class="css-79elbk"
    >
      <textarea
        class="chakra-textarea css-0"
        data-testid="textareaTest"
      />
    </div>
  </label>
</DocumentFragment>
`;
